config/loader.go

```diff
@@ -85,3 +114,95 @@ func loadConfig(cfg interface{}, fs afero.Fs, dirPath string) error {
 	l := config.NewLoader(opts...)
 	return l.Load(cfg)
 }
+
+// LoadClientConfig load the project specific config from these locations:
+// 1. filepath. ./optimus <client_command> -c "path/to/config/optimus.yaml"
+// 2. current dir. Optimus will look at current directory if there's optimus.yaml there, use it
+func LoadClientConfig(filePath string) (*ClientConfig, error) {
+	cfg := &ClientConfig{}
+
+	// getViperWithDefault + SetFs
+	v := viper.New()
+	v.SetFs(FS)
+
+	opts := []config.LoaderOption{
+		config.WithViper(v),
+		config.WithName(DefaultFilename),
+		config.WithType(DefaultFileExtension),
+	}
+
+	// load opt from filepath if exist
+	if filePath != EmptyPath {
+		if err := validateFilepath(FS, filePath); err != nil {
+			return nil, err // if filepath not valid, returns err
+		}
+		opts = append(opts, config.WithFile(filePath))
+	} else {
+		// load opt from current directory
+		opts = append(opts, config.WithPath(currPath))
+	}
+
+	// load the config
+	l := config.NewLoader(opts...)
+	if err := l.Load(cfg); err != nil {
+		return nil, err
+	}
+
+	cfg.Log.Level = LogLevel(strings.ToUpper(string(cfg.Log.Level)))
+
+	return cfg, nil
+}
+
+// LoadServerConfig load the server specific config from these locations:
+// 1. filepath. ./optimus <server_command> -c "path/to/config.yaml"
+// 2. env var. eg. OPTIMUS_SERVE_PORT, etc
+// 3. executable binary location
+// 4. home dir
```

Comment:
Let's not load ServerConfig from binary and home dir, use only flag or the environment variables.

Environment like docker might not have a home directory for non-root users.

Reply:
I thought we have discussion on it https://github.com/odpf/optimus/issues/170, to load `server config` from default location (?)

Reply:
removing `home_dir` based on our discussion 🙏